guidelines/terms/20/programmatically-determined-link-context.html

```diff
@@ -5,8 +5,8 @@
    </p>
    
    <aside class="example"><p>In HTML, information that is programmatically determinable from a link in English
-      includes text that is in the same paragraph, list, or table cell as the link or in
-      a table header cell that is associated with the table cell that contains the link.
+      includes text that is in the same <a href="https://html.spec.whatwg.org/multipage/dom.html#paragraph">paragraph</a>, 
+      list item, or table cell as the link or in a table header cell that is associated with the table cell that contains the link.
```

Comment:
I don't think WCAG should link to an evergreen HTML spec. These links can and do change. We shouldn't put ourselves in a position where we have to republish WCAG because WHATWG updated a definition.

Reply:
The link has been stable for [10 years](https://web.archive.org/web/20140901000000*/https://html.spec.whatwg.org/multipage/dom.html#paragraph), and if the paragraph in HTML were tweaked, wouldn't we want to use the updated version?

Reply:
The problem isn't the definition changing. It's the link changing. WHATWG makes no guarantee about the location of their definitions never changing. ACT has had to update rules because HTML definitions were moved. It happens. That's an easy fix when this is in an understanding document. Not so much for a normative document. This isn't the first link to WHATWG in WCAG, but the more we do this, the greater the chance of a broken in the spec at some point in the future.

Reply:
the html standard is evergreen.  you can point to a snapshot of it, but you'll get a message like this:
>This document contains the contents of the standard as of the [9025fb30a05b2275016909bc782925d2855a9a03 commit](https://github.com/whatwg/html/commit/9025fb30a05b2275016909bc782925d2855a9a03), and should only be used as a historical reference. This commit may not even have been merged into the main branch.

>  
>Do not attempt to implement this version of the standard. Do not reference this version as authoritative in any way. Instead, see https://html.spec.whatwg.org/ for the living standard.

this seems worse to me than linking to the actual specification.  more and more w3c specs are also becoming living standards.  it seems maybe a conversation between the spec editors to make sure links can remain stable is a potential avenue to consider, rather than not linking to the HTML spec.?

Reply:
No it hasn't. I for one intend to object to any proposal that would result in having to update the million copies of WCAG annually. I'm not going to count on that for a solution. I think an agreement with WHATWG that certain links will never be changed could work, but such an agreement doesn't exist.

Probably the easiest solution would be for this to link to a W3C page that redirects to WHATWG. If the URL ever changes, updated where the redirect goes. That can be done without any change to WCAG.

Reply:
@WilcoFiers To fix a link we could do an in-place change. This would not result in a change to the date of the document. There would be a note included in the Status section of the doc and the document title would get an addition to note the change. In terms of Process this is not arduous. Given the low likelihood of the WHATWG URI changing would this be acceptable?

Your idea for having a redirect is technically feasible. I would need to check what Webmaster would say about that since it is allowing for a change in a REC document without the change being documented in the document. Basically, in-place change without the Status note.

Reply:
@iadawn Linking to an evergreen spec always comes with the risk of an undocumented change. I think that's okay for an example. I think a redirect is a better solution here, since this note is going to end up being copied, link and all, into other places. With a redirect you prefer those copies from having issues too.